pkg/frontend/frontend.go

```diff
@@ -217,6 +217,18 @@ func (f *frontend) authenticatedRoutes(r *mux.Router) {
 
 	s.Methods(http.MethodPost).HandlerFunc(f.postOpenShiftClusterKubeConfigCredentials).Name("postOpenShiftClusterKubeConfigCredentials")
 
+	s = r.
```

Comment:
I see https://github.com/Azure/ARO-RP/pull/2243/files#r919542725 - there are only 2 classes. Customer API or SRE API. If this is SRE API then it is an admin API

Reply:
I would label them customer facing APIs - Really these endpoints will only be called from the Azure portal in order to enable the "Diagnose and solve problems" blade for ARO clusters. Here is an example from an AKS cluster:

![image](https://user-images.githubusercontent.com/69012315/180839306-ad262af2-e465-4ae4-8827-b2e66234ad1e.png)

Reply:
if this is part of a versioned api release this needs to be planned and coordinated with our releases. we have one coming in september, and then likely a q4 release

Reply:
Understood - I don't believe this needs to be part of a versioned api release. The AKS RP does take in an api-version parameter for this call (as you can see from the screenshot), but I do not see it passed to AppLens in any way. I also don't see an api version mentioned in the AppLens docs. To me this indicates that the AppLens calls do not need to be associated with an API version.

@weinong am I overlooking anything here?

Reply:
API is proxied through ARM and it will require ARM manifest with swagger spec

Reply:
@ross-bryan swagger change is not required as this is not customer facing api. At least I've not seen anyone doing that.

Reply:
@zacharyljones that being said, we need to add these two apis to https://docs.microsoft.com/en-us/rest/api/openshift/operations/list?tabs=HTTP

Reply:
Thanks @weinong, TIL!

Reply:
@weinong I believe I would add these two apis to the [getOperations endpoint](https://github.com/Azure/ARO-RP/blob/master/pkg/frontend/operations_get.go). I'm assuming the `Origin` field for each of these operations would be `system` because they aren't for end users. @ross-bryan is that understanding of the `Origin` field correct?

Reply:
we use system,user in AKS as this api is called by user identity from portal

Reply:
@weinong I added a `detectors/read` entry to the `operations` endpoint, similar to AKS.

@ross-bryan Should I be able to hit the `operations` endpoint locally?

![image](https://user-images.githubusercontent.com/69012315/181307888-ec2aeea0-12f7-4a74-bcc3-f89cad1cc777.png)

Reply:
@zacharyljones   requires api version https://github.com/Azure/ARO-RP/blob/master/pkg/frontend/frontend.go#L306
```
curl -k \
  "https://localhost:8443/providers/Microsoft.RedHatOpenShift/operations?api-version=2020-04-30"
  ```